### PR TITLE
reindex cleanup I

### PR DESCRIPTION
**Pull request by snopoke**
@emord 

cc @benrudolph @czue 

:fish: review by commit

### CODE REVIEW

File: corehq/form_processor/tests/test_reindex_accessors.py [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+        docs = self._get_docs(self.middle)
+        self.assertEqual(4, len(docs))
+        self.assertEqual(set(self._get_doc_ids(docs)),
+                         set(self.second_batch))
+
+        self.assertEqual(0, len(self._get_docs(self.end)))
+
+
+
+class BaseUnshardedAccessorMixin(object):
```
Comment (dimagimon) [inline corehq/form_processor/tests/test_reindex_accessors.py]: E303 too many blank lines (3)

File: corehq/form_processor/tests/test_reindex_accessors.py [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+        self.assertEqual(self._get_doc_ids(docs), self.first_batch[:2])
+
+    def test_last_doc_pk(self):
+        docs = self._get_docs(self.middle, limit=2)
+        self.assertEqual(self._get_doc_ids(docs), self.second_batch[:2])
+
+        last_doc = self.reindex_accessor().get_doc(self.second_batch[0])
+
+        docs = self._get_docs(self.middle, last_doc_pk=last_doc.pk, limit=2)
+        self.assertEqual(self._get_doc_ids(docs), self.second_batch[1:3 ])
```
Comment (dimagimon) [inline corehq/form_processor/tests/test_reindex_accessors.py]: E202 whitespace before ']'

File: corehq/form_processor/tests/test_reindex_accessors.py [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+
+
+
+class BaseShardedAccessorMixin(object):
```
Comment (dimagimon) [inline corehq/form_processor/tests/test_reindex_accessors.py]: E303 too many blank lines (3)

File: corehq/form_processor/tests/test_reindex_accessors.py [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+        return all_docs
+
+
+class BaseCaseReindexAccessorTest(BaseReindexAccessorTest):
+    accessor_class = CaseReindexAccessor
+
+    @classmethod
+    def _create_docs(cls, count):
+        case_ids = [uuid.uuid4().hex for i in range(count)]
+        forms = [create_form_for_test(cls.domain, case_id=case_id) for case_id in case_ids]
```
Comment (dimagimon) [inline corehq/form_processor/tests/test_reindex_accessors.py]: F841 local variable 'forms' is assigned to but never used

File: corehq/form_processor/tests/test_reindex_accessors.py [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+        self.assertEqual(self._get_doc_ids(docs), self.second_batch[:2])
+
+        last_doc = self.accessor_class().get_doc(self.second_batch[0])
+
+        docs = self._get_docs(self.middle, last_doc_pk=last_doc.pk, limit=2)
+        self.assertEqual(self._get_doc_ids(docs), self.second_batch[1:3])
+
+
+
+class BaseShardedAccessorMixin(object):
```
Comment (dimagimon) [inline corehq/form_processor/tests/test_reindex_accessors.py]: E303 too many blank lines (3)

File: corehq/form_processor/tests/test_reindex_accessors.py [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+        super(BaseReindexAccessorTest, cls).setUpClass()
+        cls.domain = uuid.uuid4().hex
+        # since this test depends on the global form list just wipe everything
+        FormProcessorTestUtils.delete_all_sql_forms()
+        FormProcessorTestUtils.delete_all_v2_ledgers()
+        FormProcessorTestUtils.delete_all_sql_cases()
+
+        cls.first_batch = cls._get_doc_ids(cls._create_docs(4))
+        cls.middle = datetime.utcnow()
+        time.sleep(.02)
```
Comment (benrudolph) [inline corehq/form_processor/tests/test_reindex_accessors.py]: nit: actually patching the `datetime` object could make this more reliable for different people's computer. there are some examples in other tests here: https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/reports/tests/test_daterange.py#L31

File: corehq/form_processor/backends/sql/dbaccessors.py [hunk 7]
```diff
@@ -374,38 +374,6 @@ def _get_form_ids_for_user(domain, user_id, is_deleted):
             results = fetchall_as_namedtuple(cursor)
             return [result.form_id for result in results]
 
-    @staticmethod
-    def get_all_forms_received_since(received_on_since=None, chunk_size=500):
```
Comment (benrudolph) [inline corehq/form_processor/backends/sql/dbaccessors.py]: how come we can remove all these?

**Comment by benrudolph** [comment]
👍

File: corehq/form_processor/backends/sql/dbaccessors.py (hunk 7 above)
Comment (snopoke) [inline corehq/form_processor/backends/sql/dbaccessors.py]: They've been moved to special `ReindexAccessor` classes: https://github.com/dimagi/commcare-hq/blob/master/corehq/form_processor/backends/sql/dbaccessors.py#L68

**Comment by czue** [comment]
build fail maybe related?
```
======================================================================
FAIL: test_iteration (corehq.util.tests.test_doc_processor.CaseResumableSqlModelIteratorTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/commcare-hq-ro/corehq/util/tests/test_doc_processor.py", line 214, in test_iteration
    self.assertEqual([doc["_id"] for doc in self.itr], self.all_doc_ids)
AssertionError: Lists differ: [u'59f4e30ed971465bbaf912d6fe5... != [u'59f4e30ed971465bbaf912d6fe5...
First differing element 5:
0984341bda17434b8e59549e622fcef6
6ce4ef58d5e548b08eb973e0d5dc7330
  [u'59f4e30ed971465bbaf912d6fe589499',
   u'0e684a23ce2b427db90d47eaba84399a',
   u'15c3376318cc4820adadecdf6acc6d51',
   u'e94dd704fa3a4a4a85e58f67e469785e',
   u'3be7dbc4c7ab43cc9e1758b6fe4d5703',
+  u'6ce4ef58d5e548b08eb973e0d5dc7330',
   u'0984341bda17434b8e59549e622fcef6',
   u'8632996cebf04ed39cfe114b380ab7d8',
-  u'd251202f481248c79f825ab702ef784f',
?                                     ^
+  u'd251202f481248c79f825ab702ef784f']
?                                     ^
-  u'6ce4ef58d5e548b08eb973e0d5dc7330']
-------------------- >> begin captured logging << --------------------

```

**Comment by snopoke** [comment]
Yea, its cause the sharded setup doesn't return them in the same order.
Going to remove the tests from the sharded build for now.

**Comment by emord** [comment]
tests didn't run after your latest rebase due to `[ci skip]`

**Pull request merged**